lib/Mir.ml

```diff
@@ -74,15 +128,127 @@ and ('e, 's) statement =
+  | Correlation -> (angle_brackets Fmt.string) ppf "corr_matrix"
+  | Covariance -> (angle_brackets Fmt.string) ppf "cov_matrix"
+
+let rec pp_sizedtype pp_e ppf (st, trans) =
+  match st with
+  | Ast.SInt -> Fmt.pf ppf {|%s%a|} "int" (pp_transformation pp_e) trans
+  | Ast.SReal -> Fmt.pf ppf {|%s%a|} "real" (pp_transformation pp_e) trans
+  | Ast.SVector expr ->
+      Fmt.pf ppf {|vector%a%a|} (pp_transformation pp_e) trans
+        (Fmt.brackets pp_e) expr
+  | Ast.SRowVector expr ->
+      Fmt.pf ppf {|row_vector%a%a|} (pp_transformation pp_e) trans
+        (Fmt.brackets pp_e) expr
+  | Ast.SMatrix (d1_expr, d2_expr) ->
+      Fmt.pf ppf {|matrix%a%a|} (pp_transformation pp_e) trans
+        Fmt.(pair ~sep:comma pp_e pp_e |> brackets)
+        (d1_expr, d2_expr)
+  | Ast.SArray (st, expr) ->
+      Fmt.pf ppf {|array%a%a|} (pp_transformation pp_e) trans
+        Fmt.(
+          pair ~sep:comma
+            (fun ppf st -> pp_sizedtype pp_e ppf (st, Ast.Identity))
+            pp_e
+          |> brackets)
+        (st, expr)
+
+let rec pp_statement pp_e pp_s ppf = function
+  | Assignment (assignee, expr) ->
+      Fmt.pf ppf {|@[<h>%a :=@ %a;@]|} pp_e assignee pp_e expr
+  | TargetPE expr ->
+      Fmt.pf ppf {|@[<h>%a +=@ %a;@]|} pp_keyword "target" pp_e expr
+  | NRFunApp (name, args) ->
+      Fmt.pf ppf {|@[%s%a;@]|} name Fmt.(list pp_e ~sep:comma |> parens) args
+  | Break -> pp_keyword ppf "break;"
+  | Continue -> pp_keyword ppf "continue;"
+  | Skip -> pp_keyword ppf "skip;"
+  | Return (Some expr) -> Fmt.pf ppf {|%a %a;|} pp_keyword "return" pp_e expr
+  | Return _ -> pp_keyword ppf "return;"
+  | IfElse (pred, s_true, Some s_false) ->
+      Fmt.pf ppf {|@[<v2>@[%a(%a)@] {@;%a@]@;@[<v2>@[} %a@] {@;%a@]@;}|}
+        pp_builtin_syntax "if" pp_e pred pp_s s_true pp_builtin_syntax "else"
+        pp_s s_false
+  | IfElse (pred, s_true, _) ->
+      Fmt.pf ppf {|@[<v2>@[%a(%a)@] {@;%a@]@;}|} pp_builtin_syntax "if" pp_e
+        pred pp_s s_true
+  | While (pred, stmt) ->
+      Fmt.pf ppf {|@[<v2>@[%a(%a)@] {@;%a@]@;}|} pp_builtin_syntax "while" pp_e
+        pred pp_s stmt
+  | For {loopvar; lower; upper; body} ->
+      Fmt.pf ppf {|@[<v2>@[%a(%s in %a:%a)@] {@;%a@]@;}|} pp_builtin_syntax
+        "for" loopvar pp_e lower pp_e upper pp_s body
+  | Block stmts -> Fmt.pf ppf {|@[<v>%a@]|} Fmt.(list pp_s ~sep:Fmt.cut) stmts
+  | SList stmts -> Fmt.(list pp_s ~sep:Fmt.cut |> vbox) ppf stmts
+  | Decl {decl_adtype; decl_id; decl_type} ->
+      Fmt.pf ppf {|%a%a %s;|} pp_autodifftype decl_adtype (pp_sizedtype pp_e)
+        (decl_type, Ast.Identity) decl_id
+  | FunDef {fdrt; fdname; fdargs; fdbody} -> (
+    match fdrt with
+    | Some rt ->
+        Fmt.pf ppf {|@[<v2>%a %s%a {@ %a@]@ }|} pp_unsizedtype rt fdname
+          Fmt.(list pp_fun_arg_decl ~sep:comma |> parens)
+          fdargs pp_s fdbody
+    | _ ->
+        Fmt.pf ppf {|@[<v2>%s%a {@ %a@]@ }|} fdname
```

Comment:
Don't you want to print "void" or something here? or

Reply:
Sure; that will also simplify the pretty printing logic since I can just default the return type to `void` when it is `None`.